File: annexe8B/app/src/main/java/com/example/annexe8b/MainActivity.java
package com.example.annexe8b;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    Button gauche_droite, animation_titre, splash;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        gauche_droite = findViewById(R.id.gauche_droite);
        animation_titre = findViewById(R.id.animation_titre);
        splash = findViewById(R.id.splash);

        Ecouteur ec = new Ecouteur(MainActivity.this);
        gauche_droite.setOnClickListener(ec);
        animation_titre.setOnClickListener(ec);
        splash.setOnClickListener(ec);


        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }

    public class Ecouteur implements View.OnClickListener {

        private Context context;

        // Constructor to accept the context and any views
        public Ecouteur(Context context) {
            this.context = context;
        }

        @Override
        public void onClick(View v) {

            if(v == gauche_droite){
                Intent intent = new Intent(context, Gauche_droite.class);
                context.startActivity(intent);
            }

            else if (v == animation_titre){
                Intent intent = new Intent(context, Animation_titre.class);
                context.startActivity(intent);

            }
            else if (v == splash) {
                Intent intent = new Intent(context, Splash.class);
                context.startActivity(intent);

            }

        }
    }



}

File: tp3/app/src/main/java/com/example/tp1/MainActivity.kt
package com.example.tp1

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    lateinit var ajouter: Button
    lateinit var afficher: Button
    lateinit var quitter: Button

    var ec: Ecouteur? = null

    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.enableEdgeToEdge()
        setContentView(R.layout.acceuil)
        ec = Ecouteur()
        ajouter = findViewById(R.id.ajouter)
        afficher = findViewById(R.id.afficher)
        quitter = findViewById(R.id.quitter)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v: View, insets: WindowInsetsCompat ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        ajouter.setOnClickListener(ec)
        afficher.setOnClickListener(ec)
        quitter.setOnClickListener(ec)

        Singletonmemos.getInstance(applicationContext).deserializerListe();
    }

    override fun onStop() {
        super.onStop()
        // on veut sérialiser la liste ds un fichier pour la récpérer quand on va revenir
        try{
            Singletonmemos.getInstance(applicationContext).serializerListe();
        }
        catch (e:Exception){
            e.printStackTrace()
        }


    }

    inner class Ecouteur : View.OnClickListener {
        override fun onClick(event: View) {
            val i: Intent
            // il faut que ce soit le meme object (this et equal)
            // == est un equal tandit que === est pour deux reference au meme object
            if (event == ajouter) {
                i = Intent(this@MainActivity, AjouterActivity::class.java)
                startActivity(i)
            } else if (event == afficher) {
                i = Intent(this@MainActivity, AfficherActivity::class.java)
                startActivity(i)
            } else {
                System.exit(0)
            }
        }
    }
}